### PR TITLE
Fix Part of #4938: Introduce Onboarding profile type screen

### PR DESCRIPTION
**Pull request by adhiamboperes**
## Explanation
Fixes Part of #4938: Introduce Onboarding profile type screen

Adds the profile type selection screen.

The new "I am a student" flow is not implemented, and tests have been set up to fail when the flow has been implemented.

The "I am a parent/teacher" flow has been set up to route to the existing profile screen.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

All tests pass on Espresso
Screenshot 2024-05-23 at 18 06 46

## For UI-specific PRs only
||||
| --- | --- | --- |
|| Portrait | Landscape |
| Mobile Light Mode |Screenshot_1719762068|Screenshot_1719762228|
| Mobile Dark Mode |Screenshot_1719761957|Screenshot_1719762205|
| Tablet Light Mode |Screenshot_1719762364|Screenshot_1719762334|
| Tablet Dark Mode |Screenshot_1719762388|Screenshot_1719762309|

### CODE REVIEW

File: app/src/main/AndroidManifest.xml
Comment (adhiamboperes) [inline app/src/main/AndroidManifest.xml]: Fix formatting

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt]: Replace with apply {}

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingFragmentPresenter.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingFragmentPresenter.kt]: Replace with apply

**Review by adhiamboperes — COMMENTED** [review]
Self- review pass

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivityPresenter.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivityPresenter.kt]:
```suggestion
      ).commitNow()
```

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt]:
```suggestion
    }
    
    return binding.root
```

File: app/src/main/res/drawable/learner_otter.png
Comment (theMr17) [inline app/src/main/res/drawable/learner_otter.png:1]: This and the other added image seems to be of quite low resolution when viewed on a device (Also can be seen in screenshots). Are these final images or will these replaced later?

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivity.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivity.kt]:
```suggestion
        decorateWithScreenName(ONBOARDING_PROFILE_TYPE_ACTIVITY)
```
Please import the screen name directly.

File: app/src/main/res/values/color_defs.xml
Comment (theMr17) [inline app/src/main/res/values/color_defs.xml:144]: Why is this and the corresponding change in `color_palette.xml` file required in this PR?

File: app/src/main/res/layout/onboarding_profile_type_activity.xml
Comment (theMr17) [inline app/src/main/res/layout/onboarding_profile_type_activity.xml]: Can we simplify the file to contain only the `FrameLayout` (`layout` and `LinearLayout` can be removed)?

File: app/src/main/res/layout/onboarding_profile_type_fragment.xml
Comment (theMr17) [inline app/src/main/res/layout/onboarding_profile_type_fragment.xml]: Please create a custom style for this button. It will help us in getting rid of the duplicate code in the different qualifier layout files of this fragment.

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (theMr17) [inline app/src/main/res/layout-land/onboarding_profile_type_fragment.xml:1]: The file has an extra indentation from line 5 to 136.

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (theMr17) [inline app/src/main/res/layout-land/onboarding_profile_type_fragment.xml]: Please move the profile type title text (in mobile landscape layout only) closer to the cards to match the positioning in the [mocks](https://www.figma.com/file/PnCtQKDR5VWbIlA3Ui7rTa/Oppia---Onboarding-Android-App?type=design&node-id=1955-38502&mode=design&t=qzqNLMUiohEg63xu-4) more closely.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]:
```suggestion
      onView(withId(R.id.profile_type_title))
        .check(matches(isDisplayed()))
        .check(
          matches(
            withText(
              R.string.onboarding_profile_type_activity_header
            )
          )
        )
```

Here and other tests where it is checked that the view is displayed and then checked for the content of the view. We can simplify it as shown above.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Please add test for landscape orientation as well.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Please add test for landscape orientation as well.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Please add test for landscape orientation as well.

**Review by theMr17 — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! I have left a few comments.

File: app/src/main/res/values/color_palette.xml
Comment (XichengSpencer) [inline app/src/main/res/values/color_palette.xml:1]: I also want to know the reason why we remove the 'survey' part. Our color naming in color_palette.xml seems a bit confused to me. Some contain functional descriptions like 'survey' or 'onborading' while others don't. But since falls in the Survey section in the color palette, I think it will be good to keep the survey part.

File: app/src/main/res/layout-sw600dp-land/onboarding_profile_type_fragment.xml
Comment (XichengSpencer) [inline app/src/main/res/layout-sw600dp-land/onboarding_profile_type_fragment.xml:22]: I wonder how we set those guideline percentages?
It seems there are various percentages on different screens, is it necessary to set a standard for them to keep a consistent look?

Screenshot 2024-04-16 153147

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (XichengSpencer) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: I am not sure if every test should annotate the framework it is using

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (XichengSpencer) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: It confuses me when previous comments in Espresso tests say "// Robolectric is usually not used to test the interaction of
  // Android components"
  To me perform(click()) seems like an interaction

**Review by XichengSpencer — COMMENTED** [review]
Thank you Adhiambo and MR17, 
Those are great works! I have left some thoughts in addition to MR17's comments, other than that LGTM!

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivityPresenter.kt
Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivityPresenter.kt]: Done

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt
Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt]: This gave a ktlint error:
`android/app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeFragmentPresenter.kt:39:1: Trailing space(s)`
So I have kept it as is.

File: app/src/main/res/drawable/learner_otter.png
Comment (adhiamboperes) [reply app/src/main/res/drawable/learner_otter.png:1]: At the moment, these are everything we have from figma. We could talk to the art team about getting better resolution images as part of pre-release efforts.

File: app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivity.kt
Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/onboardingv2/OnboardingProfileTypeActivity.kt]: Done

File: app/src/main/res/values/color_defs.xml
Comment (adhiamboperes) [reply app/src/main/res/values/color_defs.xml:144]: This is simply an incidental improvement that I made, since it is minor and I didn't want to file an issue/create a seperate PR for it.

File: app/src/main/res/layout/onboarding_profile_type_activity.xml
Comment (adhiamboperes) [reply app/src/main/res/layout/onboarding_profile_type_activity.xml]: Yes, this seems to work well enough.

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout-land/onboarding_profile_type_fragment.xml:1]: Fixed

File: app/src/main/res/layout/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout/onboarding_profile_type_fragment.xml]: Done

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout-land/onboarding_profile_type_fragment.xml]: Done, please see lates screenshots.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Thanks for the suggestion. I have used the `allOf` check instead as follows:
```kotlin
  onView(withId(R.id.profile_type_title))
        .check(
          matches(
            allOf(
              isDisplayed(),
              withText(
                R.string.onboarding_profile_type_activity_header
              )
            )
          )
        )
```

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: In landscape we don't have this stepcount view, so there is no need to write a test.

File: app/src/main/res/layout-sw600dp-land/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout-sw600dp-land/onboarding_profile_type_fragment.xml:22]: At the moment, we don't have any defined standards regarding guidelines. So it is at the developer's discretion to use what will make the UI look closest to the mocks. We can however open a discusion with the team regarding this.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: If it is not annotated with the patform, they will run on both. If it is annotated, it means that it will only run on the specified platform, mainly because there is some sort of issue with running on the alternate platform.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: By android components I meant things like backpress/back button. I realized these are not very good tests, because we don't typically test the android lifecycle(here, I wanted to know wether an activity will finish and go back to the previous one). I have therefore removed these tests.

**Comment by adhiamboperes** [comment]
Thanks @theMr17, @XichengSpencer! I addressed your comments. Please look them over and let me know!

File: app/src/main/res/values/color_palette.xml
Comment (adhiamboperes) [reply app/src/main/res/values/color_palette.xml:1]: Per https://github.com/oppia/oppia-android/wiki/Dark-Mode, we should not name colors in `color_defs.xml` after features.

I agree, the color files are quite disorganized, and we will need to refactor it at some point.

**Review by theMr17 — APPROVED** [review]
Thanks @adhiamboperes, this LGTM! Please feel free to mark the comments as resolved.

File: app/src/main/res/drawable/learner_otter.png
Comment (BenHenning) [inline app/src/main/res/drawable/learner_otter.png:1]: For this & the other PNG--can these also be SVGs, instead?

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]:
```suggestion
  <string name="onboarding_profile_type_activity_parent_text">I\'m the parent, teacher, or guardian of a student.</string>
```

Nit: Oxford comma

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Do this & the previous test actually need to be constrained to Espresso?

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt:113]: Any tests for this or the activity to cover back navigation?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! Had a few comments--PTAL.

Also per my review of the blocking PR, I wonder if we could just keep everything under onboarding/ rather than having a separate package. WDYT?

File: app/src/main/res/drawable/learner_otter.png
Comment (adhiamboperes) [reply app/src/main/res/drawable/learner_otter.png:1]: For these specific images, we do not have svg versions.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [reply app/src/main/res/values/strings.xml]: Added

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Not specifically. I have removed the restriction.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt:113]: Given that back button here finishes the current activity, I have added tests to check that activity is actually destroyed.

**Comment by adhiamboperes** [comment]
@BenHenning, PTAL.

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/onboarding_profile_type_fragment.xml]: Can this be flattened? I think most (all?) nested ConstraintLayouts should theoretically be simplifiable (except when separated by a non-constraint layout like MaterialCardView below).

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Alternatively, you could verify that the activity is finishing using Activity.isFinishing() which I think is Robolectric-friendly.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt:113]: Perfect--thanks!

File: app/src/main/res/drawable/learner_otter.png
Comment (BenHenning) [reply app/src/main/res/drawable/learner_otter.png:1]: I wonder if they have source materials (e.g. Illustrator) that can be used to export SVG versions. The concern with using PNGs (aside from the binary implications in the repository) is that they won't scale correctly for different display resolutions without introducing multiple copies of the PNGs. SVGs scale arbitrarily and should always look correct.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! Mostly LGTM, but had a few comments--PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt]: Thanks,`Activity.isFinishing()` does run on robolectric.

File: app/src/main/res/layout-land/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout-land/onboarding_profile_type_fragment.xml]: Done for all the alternate layouts

**Comment by adhiamboperes** [comment]
@BenHenning, I have made the requested changes to the layout and the tests.

Re -  svg assets, I have checked with the art team, waiting for a response. Although the original designer(Julia) mentioned that she personally does not have them.

File: app/src/main/res/layout-sw600dp-port/onboarding_profile_type_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-port/onboarding_profile_type_fragment.xml]: Is this indented correctly? Seems like it needs a reformatting.

**Review by BenHenning — APPROVED** [review]
Thanks @adhiamboperes! Just had one follow-up nit--PTAL. Approving the PR so feel free to resolve the remaining 2 comments once they're addressed on your end.

For the comment specifically on SVGs, I think we do want to try and address this properly before merging the PR for the reasons I mentioned--the PNGs will either require too much resources for smaller displays or will look fuzzy on larger displays.

File: app/src/main/res/drawable/learner_otter.png
Comment (adhiamboperes) [reply app/src/main/res/drawable/learner_otter.png:1]: Molly from the art team is working on converting the pngs to svgs. She could not find a repository of source materials.

File: app/src/main/res/layout-sw600dp-port/onboarding_profile_type_fragment.xml
Comment (adhiamboperes) [reply app/src/main/res/layout-sw600dp-port/onboarding_profile_type_fragment.xml]: Reformatted the file

File: app/src/main/res/drawable/learner_otter.png
Comment (adhiamboperes) [reply app/src/main/res/drawable/learner_otter.png:1]: Received the images and replaced them here.

Comment (adhiamboperes) [reply app/src/main/res/drawable/learner_otter.png:1]: Just FYI that I replaced these images and updated the screenshots.

**Review by adhiamboperes — COMMENTED** [review]
Reviewed latest changes, looks good.

**Comment by adhiamboperes** [comment]
Resolved remaining comments, and enabling auto-merge

**Pull request merged**
